browser/resources/settings/brave_search_engines_page/brave_search_engines_page.ts

```diff
@@ -48,20 +68,47 @@ class BraveSearchEnginesPage extends BraveSearchEnginesPageBase {
       'private-search-engines-changed', updatePrivateSearchEngines)
   }
 
-  shouldShowSearchSuggestToggle_() {
+  private onOpenPrivateDialogButtonClick_() {
+    this.showPrivateSearchEngineListDialog_ = true
+  }
+
+  private onPrivateSearchEngineListDialogClose_() {
+    this.showPrivateSearchEngineListDialog_ = false
+  }
+
+  private onPrivateSearchEngineChangedInDialog_(e: CustomEvent) {
+    this.confirmationToastLabel_ = this.i18n(
+      'privateSearchEnginesConfirmationToastLabel', e.detail.searchEngine.name)
+    this.shadowRoot!.querySelector<CrToastElement>(
+      '#confirmationToast')!.show()
+  }
+
+  private shouldShowSearchSuggestToggle_() {
     return !loadTimeData.getBoolean('isGuest')
   }
 
-  shouldShowPrivateSearchProvider_(prefs) {
+  private shouldShowPrivateSearchProvider_(
+    prefs: chrome.settingsPrivate.PrefObject)
+  {
     // When default search engine is enforced, configured provider is not used.
     // If we install search provider extension, that extension will be used on normal and
     // private(tor) window. So, just hide this option.
     return !loadTimeData.getBoolean('isGuest') && !this.isDefaultSearchEngineEnforced_(prefs)
   }
 
-  isDefaultSearchEngineEnforced_(prefs) {
-    return prefs.enforcement === chrome.settingsPrivate.Enforcement.ENFORCED;
+  private isDefaultSearchEngineEnforced_(
+    prefs: chrome.settingsPrivate.PrefObject)
+  {
+    return prefs.enforcement === chrome.settingsPrivate.Enforcement.ENFORCED
+  }
+
+  private computeDefaultPrivateSearchEngine_() {
+    if (!this.privateSearchEngines_.length) {
+      return null
+    }
+
+    return this.privateSearchEngines_.find(engine => engine.default)!
```

Comment:
Do we need to return `null` here, rather than undefined?

If not then
```suggestion
    return this.privateSearchEngines_.find(engine => engine.default)
```

should work. If it does matter then

```suggestion
    return this.privateSearchEngines_.find(engine => engine.default) ?? null
```

is slightly more idiomatic

Reply:
Used your first approach, much nicer!